devconf-ironic-and-edgy/source/index.rst

```diff
@@ -0,0 +1,380 @@
+  analyze unless one has console access to the system.
+
+The history of network booting
+==============================
+
+* PXE: BOOTP/DHCP -> TFTP
+* iPXE: BOOTP/DHCP -> HTTP/iSCSI
+* UEFI: BOOTP/DHCP -> HTTP/iSCSI
+* Virtual Media: HTTP
+
+.. Things to talk about ^
+
+  The problem of network booting has been approached long ago.
+
+  The first well-defined and established procedure to perform the booting
+  is known as *PXE*. It relies on a suite of Internet procotols of the time.
+  PXE has been designed for LANs, resource-constrained NICs and smaller-scale
+  installations. These were probably the reasons to use UDP for all the involved
+  protocols.
+
+  Over time, the choice of UDP has become a nuisance so that the *PXE*
+  successor - *iPXE* (and later *UEFI* boot loader) introduced HTTP boot
+  effectively replacing less reliable and less scalable *TFTP* for boot image
+  transfer purposes.
+
+  Still, the initial network configuration phase needs to rely on UDP-based
+  DHCP protocol. With introduction of the virtual media boot technology,
+  this last fragile piece in the boot sequence has been replaced making
+  virtual media boot nearly ideal way to boot distant computers.
+
+Redfish: virtual media
+======================
+
+Features:
+
+* Layer-3 based deployment possible
+* Ensures authentic boot image
+* Ability to cache boot images
+
+.. Things to talk about ^
+
+  With virtual media, the boot image is pulled by the BMC rather than
+  the booting system itself. Then BMC emulates a local CD drive using
+  the downloaded image. The system gets booted from this virtual CD
+  for one or more times.
+
+  It is generally more reliable and secure to let BMC pulling specific
+  boot image because BMC does not need to perform network bootstrapping.
+  With BMC it's easier to ensure boot image authenticity and consistency.
+
+  On top of that, BMC has the potential to cache and reuse boot images
+  for one or many systems what is important considering the sizes of the
+  boot images and potential connectivity constraints at the edge.
+
+  Redfish fully supports virtual media operations so it fits well with
+  the edge use-case.
+
+Virtual media Layer-3 deployment
+================================
+
+* Ironic deploy image still requires DHCP
+* Virtual Media offers virtual floppy \o/
+
+.. Things to talk about ^
+
+  There is still one step in the ironic bare metal instance deployment
+  process which requires network configuration step over DHCP. The
+  so-called deploy image (the one which pulls the installation image
+  and writes it down to the local system drive) needs DHCP thus
+  requiring either DHCP server in the broadcast domain or some form of
+  tunelling or proxying.
+
+  There has been a fairly new ironic specification proposed to use
+  virtual media floppy to pass static network configuration information
+  for the deploy image to consume.
+
+Redfish: composable hardware
```

Comment:
This does not feel like related topic to, as excited as it is.

Reply:
Proposed take on this is to treat remotely controlled hardware composition as a way to compensate for the missing sysadmin on site.

Because only polar and ironic bears can flourish beyond the Arctic Circle.

Reply:
Well, in practice a composable hardware rack is a huuuge thing. For edge we're talking about 2-3 beefy machines instead. Sometimes not even beefy.